### DISCUSSION TITLE
Generating Tailwind Configs Programatically

### DISCUSSION THREAD

**Discussion by Ghirigoro** [tailwindlabs/tailwindcss#9179] [Help] [answered]
I've got a monorepo that's designed as a kind of framework to create front-end projects. The goal is to make a lot of these front-ends and to make it really easy to spin one up, so we want to minimize the setup and configuration of each. Ideally we'd love to have a script that given a set of objects defining overrides to the base config (e.g. colors, spacing, etc.) will generate a tailwind config to be used in the build system. This way each new project only has to define a set of JSON files describing the overrides and is ready to go. Tailwind seems to require a file on disk to make this work (we could always do that if necessary) but I was wondering if we were missing something and could do this entirely in memory.

So my questions are:
- Can you build and use tailwind config files entirely in memory?
- Does anyone have any experience with building a kind of "meta" config system for Tailwind that makes it more reusable across projects that they would be willing to share?

Thanks!

P.S. - We're using Vite to build our front-ends in case that matters.

**Comment by wongjn** [comment, marked as answer]
You can use the [presets](https://tailwindcss.com/docs/presets) to have a base configuration. Then, for your Tailwind config in memory, if you are using PostCSS, you can pass a Tailwind config object into the `tailwindcss` PostCSS plugin inline, something like:

```js
const tailwindcss = require('tailwindcss');
const postcss = require('postcss');

await postcss([tailwindcss(myConfig)]).process(css);
```
You'll need to adapt this for your build system, etc but it should give you a rough idea.

**Reply by Ghirigoro** [reply]
Thanks! That helps a lot.
